session/session.go

```diff
@@ -0,0 +1,34 @@
+package session
+
+import (
+	"fmt"
+	"time"
+
+	"launchpad.net/gozk"
+)
+
+var ErrNotConnected = fmt.Errorf("Unable to connect to ZooKeeper.")
+
+type ZkSession struct {
+	Connection *zookeeper.Conn
+	Events     <-chan zookeeper.Event
+}
+
+func NewZkSession(servers string) (*ZkSession, error) {
+	zkconn, zkevents, err := zookeeper.Dial(servers, 3*time.Second)
+
+	if err != nil {
+		return nil, err
+	}
+
+	select {
+	case event := <-zkevents:
+		if event.State != zookeeper.STATE_CONNECTED {
```

Comment:
So zookeeper's connection management is... kinda subtle, afaict.

You're supposed to keep reading off of this channel, and sometimes have to reconnect. gozk will actually panic if you fail to read enough messages from this channel.

For example, if a network partition causes you to disconnect from zookeeper, you'll receive a `STATE_EXPIRED_SESSION` or something like that, and possibly some other combination of states.

I'm really not sure what the best way to do this is. Wrapping all that session management up in a `ZKSession` type is clearly a good idea, but I think you're probably going to have to be more careful about the whole lifecycle of this connection, and possibly re-establishing it via `zookeeper.Redial()`.